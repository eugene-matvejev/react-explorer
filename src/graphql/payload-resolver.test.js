import fn from './payload-resolver';

describe('form payload resolver', () => {
    it('should invoke ::valueTransformer if it been provided, with ::value as payload', () => {
        const spy = jest.fn();
        const c = [
            {
                items: [
                    {
                        attr: 'test',
                        value: 1,
                        valueTransformer: spy,
                    },
                ],
            },
        ];

        fn(c);

        expect(spy).toBeCalledWith(c[0].items[0].value);
    });

    it('should pass result of ::valueTransformer in to payload', () => {
        const c = [
            {
                items: [
                    {
                        attr: 'test',
                        value: 1,
                        valueTransformer: (v) => 2,
                    },
                ],
            },
        ];

        expect(fn(c)).toMatchSnapshot();
    });

    it(`should pass ::value 'as is' into payload`, () => {
        const c = [
            {
                items: [
                    {
                        attr: 'test',
                        value: 1,
                    },
                ],
            },
        ];

        expect(fn(c)).toMatchSnapshot();
    });

    it(`should aggregate resolved fields' values with same ::attr into array`, () => {
        const c = [
            {
                items: [
                    {
                        attr: 'test',
                        value: 1,
                        seq: 0x10,
                    }
                ],
            },
            {
                items: [
                    {
                        attr: 'test',
                        value: 1,
                        valueTransformer: (v) => 2,
                        seq: 0x10,
                    }
                ],
            },
        ];

        expect(fn(c)).toMatchSnapshot();
    });
});
