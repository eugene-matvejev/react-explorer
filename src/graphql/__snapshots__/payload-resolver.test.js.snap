// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form payload resolver should aggregate resolved fields' values with same ::attr into array 1`] = `
Object {
  "test": Array [
    1,
    2,
  ],
}
`;

exports[`form payload resolver should pass ::value 'as is' into payload 1`] = `
Object {
  "test": 1,
}
`;

exports[`form payload resolver should pass result of ::valueTransformer in to payload 1`] = `
Object {
  "test": 2,
}
`;
